lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/io/ByteBuffers.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.io;
+
+import java.nio.ByteBuffer;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public final class ByteBuffers
+{
+    private ByteBuffers() {}
+
+    /**
+     * Gets the bytes the provided {@link ByteBuffer} wraps, without advancing buffer position.
+     * Throws when provided buffer does not directly wrap bytes.
+     */
+    public static byte[] getWrappedBytes(ByteBuffer byteBuffer)
+    {
+        checkArgument(byteBuffer.hasArray(), "buffer does not have array");
+        checkArgument(byteBuffer.arrayOffset() == 0, "buffer has non-zero array offset: %s", byteBuffer.arrayOffset());
+        checkArgument(byteBuffer.position() == 0, "buffer has been repositioned to %s", byteBuffer.position());
+        checkArgument(byteBuffer.remaining() == byteBuffer.array().length, "buffer has %s remaining bytes while array length is %s", byteBuffer.remaining(), byteBuffer.array().length);
```

Comment:
nit: `buffer.array()` it is called `2` times in this statement and once again when it is returned.

Reply:
I am aware. This was supposed to be optimized for readability. Note that `array()` is a cheap accessor method.
Extracted anyway